codec/src/main/java/io/netty/handler/codec/compression/LzfEncoder.java

```diff
@@ -138,6 +176,48 @@ protected void encode(ChannelHandlerContext ctx, ByteBuf in, ByteBuf out) throws
         }
     }
 
+    private int encodeCompress(byte[] input, int inputPtr, int length, byte[] output, int outputPtr) {
+        return LZFEncoder.appendEncoded(encoder,
+                input, inputPtr, length, output, outputPtr) - outputPtr;
+    }
+
+    /**
+     * Use lzf uncompressed format to encode a piece of input.
+     */
+    private static int encodeNonCompress(byte[] input, int inputPtr, int length, byte[] output, int outputPtr) {
+        int left = length;
+        int chunkLen = Math.min(LZFChunk.MAX_CHUNK_LEN, left);
+        outputPtr = LZFChunk.appendNonCompressed(input, inputPtr, length, output, outputPtr);
+        left -= chunkLen;
+        if (left < 1) {
+            return outputPtr;
+        }
+        inputPtr += chunkLen;
+        do {
+            chunkLen = Math.min(left, LZFChunk.MAX_CHUNK_LEN);
+            outputPtr = LZFChunk.appendNonCompressed(input, inputPtr, length, output, outputPtr);
+            inputPtr += chunkLen;
+            left -= chunkLen;
+        } while (left > 0);
+        return outputPtr;
+    }
+
+    public int getCompressThreshold() {
+        return compressThreshold;
+    }
+
+    /**
+     * Since we could set this threshold at runtime, so we keep set method.
+     */
+    public void setCompressThreshold(int compressThreshold) {
```

Comment:
I would prefer to not allow this for now... I think there is really no good reason to adjust this on the fly.